### DISCUSSION TITLE
Display Additional attribute from json file

### DISCUSSION THREAD

**Discussion by Axons-glitch** [oceanofthelost/InteractivePCB#105] [Q&A] [answered]
Hi, I really like this project.
I just modified the ulp file so it writes the part package as a new attribute so in json file there is "package":"_here is the value_", for each component in the attribute section.  
But I can't figure out how to display the package as an attribute in HTML. The only Additional attribute that seems to have value is "name".

Thanks for any help.

**Comment by oceanofthelost** [comment, marked as answer]
With iPCB loaded in your browser, hover over gear icon in top left

![image](https://user-images.githubusercontent.com/1684123/182776684-f8b84483-e217-476e-9f90-38dc3000bfad.png)

This will bring up a new popup with some settings. At the bottom of the popup is a box labeled "Additional Attributes". 

![image](https://user-images.githubusercontent.com/1684123/182776903-47fec63e-6041-485d-a487-d650b2fee1ae.png)

Add "package" to the box as shown above. A new column will be added to the BOM table, as shown below:
![image](https://user-images.githubusercontent.com/1684123/182777085-def2fd97-56a9-4ac8-a960-3a57fe340264.png)

If a part has an attribute of "package" then package will be be displayed in this column.

**Reply by Axons-glitch** [reply]
That is exactly what I do and it does not work except for "name".

Here is one part from json.
![image](https://user-images.githubusercontent.com/110108228/182790218-2bfb40ad-423f-43c7-84e1-76ef5083b27e.png)

Here is the Additional Attribute and BOM table.
![1](https://user-images.githubusercontent.com/110108228/182791692-5893c974-087c-44dd-b074-3151f0bb5b82.png)
![2](https://user-images.githubusercontent.com/110108228/182791716-8bcf6fdf-21a9-45da-9224-4d140fb42f94.png)

When I try "name" instead of "package" it works.
![3](https://user-images.githubusercontent.com/110108228/182792295-65ff8b0c-8ddd-473b-ab02-e94b75e6d33a.png)
![4](https://user-images.githubusercontent.com/110108228/182792297-949db501-0b19-4f8a-a288-41093bf1ffe2.png)

*The part named "2K8" is a mistake on my board.

**Reply by oceanofthelost** [reply]
Attributes in in V2 use a semicolon list for key and value. 

Modify your ULP to produce the folling

```
"attributes":{
    "name":"NAME;VALUE;package",
    "value":"C21;1uF;C1206",
},
```

iPCB will then read name and value semicolon lists and parse them to create correctly formatted BOM entries.
